### PR TITLE
Feature/app insights

### PR DESCRIPTION
**Pull request by kevin11-allen-nttd**
Adds option for app insights

Adds option for azure monitor private link scope
- Creates private DNS zones:
  ```
    "privatelink.monitor.azure.com",
    "privatelink.oms.opinsights.azure.com",
    "privatelink.ods.opinsights.azure.com",
    "privatelink.agentsvc.azure-automation.net",
    "privatelink.blob.core.windows.net"
  ```
    - https://learn.microsoft.com/en-us/azure/private-link/private-endpoint-dns#management-and-governance
- Anything in the Vnet will access azure monitor over private link
  - Target azure monitor resources (app insights, log analytics) must be linked to the scope

### CODE REVIEW

**Review by debasish-sahoo-nttd — APPROVED** [review]
lets just wait for the primitive module to be merged and then use its tag instead of the feature branch before we merge this .

**Review by debasish-sahoo-nttd — APPROVED** [review]

**Review by chris11-taylor-nttd — APPROVED** [review]

**Pull request merged**
